### DISCUSSION TITLE
How do I "popTo" in react-navigation 7.x on a BottomTabBarProps['navigation'] type?

### DISCUSSION THREAD

**Discussion by samholmes** [react-navigation/react-navigation#12034] [Q&A]
My tab bar component implemented a handler with this logic that worked in react-navigation 6:

```ts
  const handleOnPress = useHandler(() => {
    switch (route.name) {
      case 'homeTab':
        return navigation.navigate('homeTab', currentName === 'homeTab' ? { screen: 'home' } : {})
      case 'walletsTab':
        return navigation.navigate('walletsTab', currentName === 'walletsTab' ? { screen: 'walletList' } : {})
      case 'buyTab':
        return navigation.navigate('buyTab', currentName === 'buyTab' ? { screen: 'pluginListBuy' } : {})
      case 'sellTab':
        return navigation.navigate('sellTab', currentName === 'sellTab' ? { screen: 'pluginListSell' } : {})
      case 'swapTab':
        return navigation.navigate('swapTab', currentName === 'swapTab' ? { screen: 'swapCreate' } : {})
      case 'extraTab':
        return navigation.navigate('extraTab')
      case 'devTab':
        return navigation.navigate('devTab')
    }
  })
```

You can see I'm relying on `.navigate` to pop to the previous screen if the tab is tapped a second time after focus. In version 7, `.popTo` is suppose to be used, but `popTo` is only available on navigations from "stack navigators", so it's not a method on the `navigation` object received by my tab component (see `BottomTabBarProps`), which is of the type `NavigationHelpers<ParamListBase, BottomTabNavigationEventMap>`.

What's the correct way to implement my tab component's behavior in react-navigation 7.x?

**Comment by satya164** [comment]
If you specify a `getId` prop on your screens then it should work as you expect.

https://reactnavigation.org/docs/screen/#getid

Alternatively you can dispatch a `popTo` action on the child navigator using it's key:

```js
navigation.dispatch({
  ...StackActions.popTo('ScreenName'),
  target: stateKey // from `route.state.key` of the screen containing the navigator
})
```
